packages/swapper/src/swappers/osmosis/OsmosisSwapper.ts

```diff
@@ -309,6 +333,7 @@ export class OsmosisSwapper implements Swapper<ChainId> {
 
     const signed = await osmosisAdapter.signTransaction(signTxInput)
     const tradeId = await osmosisAdapter.broadcastTransaction(signed)
+    const cosmosTxHistory = await cosmosAdapter.getTxHistory({ pubkey: cosmosAddress })
```

Comment:
we can add `pageSize: 1` to limit this request, and we should also probably fetch it just prior to broadcast to be extra safe.